x/ibc/light-clients/solomachine/types/proof.go

```diff
@@ -13,10 +15,30 @@ import (
 )
 
 // VerifySignature verifies if the the provided public key generated the signature
-// over the given data.
-func VerifySignature(pubKey crypto.PubKey, data, signature []byte) error {
-	if !pubKey.VerifySignature(data, signature) {
-		return ErrSignatureVerificationFailed
+// over the given data. Single and Multi signature public keys are supported.
+// The type of the signature data determines how the public key is used to
+// verify the signature. An error is returned if signature verification fails
+// or an invalid SignatureData type is provided.
+func VerifySignature(pubKey crypto.PubKey, signBytes []byte, sigData signing.SignatureData) error {
+	switch data := sigData.(type) {
+	case *signing.SingleSignatureData:
+		if !pubKey.VerifySignature(signBytes, data.Signature) {
+			return ErrSignatureVerificationFailed
+		}
+
+	case *signing.MultiSignatureData:
+		multiPK, ok := pubKey.(multisig.PubKey)
+		if !ok {
+			return sdkerrors.Wrapf(ErrSignatureVerificationFailed, "invalid pubkey type: expected %T, got %T", (multisig.PubKey)(nil), pubKey)
+		}
+		if err := multiPK.VerifyMultisignature(func(mode signing.SignMode) ([]byte, error) {
+			return signBytes, nil
+		}, data); err != nil {
```

Comment:
I don't understand this function here? also we could replace `mode` for `_`

Reply:
Ah thanks for pointing this out. It fulfills the `VerifyMultisignature` interface, I'll add a comment with more detail